uefi/src/main.rs

```diff
@@ -97,6 +98,20 @@ fn main_inner(image: Handle, mut st: SystemTable<Boot>) -> Status {
     // Ramdisk must load from same source, or not at all.
     let ramdisk = load_ramdisk(image, &mut st, boot_mode);
 
+    let config_file = load_config_file(image, &mut st, boot_mode);
+    let config = match BootloaderConfigFile::deserialize(config_file) {
+        Ok(data) => data,
+        Err((data, err)) => {
+            writeln!(
+                st.stdout(),
+                "Failed to deserialize the config file {:?}",
+                err
+            )
+            .unwrap();
+            data
+        }
+    };
```

Comment:
See above: I think this would be simpler without the `BootloaderConfigFile::deserialize` method.